addons/website/controllers/main.py

```diff
@@ -529,47 +529,54 @@ def autocomplete(self, search_type=None, term=None, order=None, limit=5, max_nb_
-                                    "website.search_text_with_highlight",
-                                    {'parts': parts}
-                                )
-                                field_type = 'html'
-
-                if field_type not in ('image', 'binary') and ('ir.qweb.field.%s' % field_type) in request.env:
-                    opt = {}
-                    if field_type == 'monetary':
-                        opt['display_currency'] = options['display_currency']
-                    value = request.env[('ir.qweb.field.%s' % field_type)].value_to_html(value, opt)
-                mapped[mapped_name] = escape(value)
-            result.append(mapped)
+            templateKey = search_result["template_key"]
+            group_title = search_result["group_title"]
+            for record in search_result['results_data']:
+                mapping = record['_mapping']
+                mapped = {
+                    '_fa': record.get('_fa'),
+                }
+                for mapped_name, field_meta in mapping.items():
+                    value = record.get(field_meta.get('name'))
+                    if not value:
+                        mapped[mapped_name] = ''
+                        continue
+                    field_type = field_meta.get('type')
+                    if field_type == 'text':
+                        if field_meta.get('truncate', True):
+                            value = shorten(value, max_nb_chars, placeholder='...')
+                        if field_meta.get('match') and term:
+                            pattern = '|'.join(map(re.escape, term.split()))
+                            if pattern:
+                                parts = re.split(f'({pattern})', value, flags=re.IGNORECASE)
+                                if len(parts) > 1:
+                                    value = request.env['ir.ui.view'].sudo()._render_template(
+                                        "website.search_text_with_highlight",
+                                        {'parts': parts}
+                                    )
+                                    field_type = 'html'
+
+                    elif field_type not in ('image', 'binary') and ('ir.qweb.field.%s' % field_type) in request.env:
+                        opt = {}
+                        if field_type == 'monetary':
+                            opt['display_currency'] = options['display_currency']
+                        elif field_type == 'float':
+                            opt['precision'] = field_meta.get('precision', 2)
+                        value = request.env[('ir.qweb.field.%s' % field_type)].value_to_html(value, opt)
+                    mapped[mapped_name] = escape(value)
+
+                # Only create group when data exist
+                if group_title not in result:
+                    result[group_title] = {
+                        "templateKey" : templateKey,
+                        "rowTitle": group_title,
```

Comment:
```suggestion
                        "groupName": group_name,
```
I chose consistency here. If it was previously called group_title, it should be the same here as well.